src/components/auction/OrderPlacement/index.tsx

```diff
@@ -262,6 +264,27 @@ const OrderPlacement: React.FC<OrderPlacementProps> = (props) => {
   ])
   const notApproved = approval === ApprovalState.NOT_APPROVED || approval === ApprovalState.PENDING
   const orderPlacingOnly = auctionState === AuctionState.ORDER_PLACING
+  const coversClearingPrice = (price: string | undefined, invertedPrice: boolean): boolean => {
+    if (!price || price === '-') {
+      return false
+    }
+    const [intPart, decimalsPart = ''] = price.split('.')
+    const decimals = decimalsPart.length
+
+    const divisor =
+      decimals > 0 ? BigNumber.from(10).pow(BigNumber.from(decimals)) : BigNumber.from(1)
+    const priceBN = BigNumber.from(intPart)
+      .mul(divisor)
+      .add(BigNumber.from(decimalsPart || '0'))
+    const fractionPrice = new Fraction(priceBN.toString(), divisor.toString())
+
+    return invertedPrice
+      ? derivedAuctionInfo.clearingPrice.invert().lessThan(fractionPrice)
+      : derivedAuctionInfo.clearingPrice.greaterThan(fractionPrice)
+  }
+  const hasRiskNotCoveringClearingPrice =
+    auctionState === AuctionState.ORDER_PLACING_AND_CANCELING &&
+    coversClearingPrice(price, showPriceInverted)
```

Comment:
looks good

though would be great to reuse the logic from here:
```
Please take a look at: useGetOrderPlacementError

You can reuse here:

const price = showPricesInverted
    ? getInverse(Number(priceFromState), NUMBER_OF_DIGITS_FOR_INVERSION).toString()
    : priceFromState

const { buyAmountScaled, sellAmountScaled } = convertPriceIntoBuyAndSellAmount(
    derivedAuctionInfo?.auctioningToken,
    derivedAuctionInfo?.biddingToken,
    price == '-' ? '1' : price,
    sellAmount,
  )
if (
  derivedAuctionInfo?.clearingPriceSellOrder != null &&
  derivedAuctionInfo?.clearingPrice != null &&
  derivedAuctionInfo?.auctioningToken != undefined &&
  derivedAuctionInfo?.biddingToken != undefined &&
  auctionState == AuctionState.ORDER_PLACING &&
  buyAmountScaled &&
  sellAmountScaled
    ?.mul(derivedAuctionInfo?.clearingPriceSellOrder?.buyAmount.raw.toString())
    .lte(
      buyAmountScaled.mul(derivedAuctionInfo?.clearingPriceSellOrder?.sellAmount.raw.toString()),
    )
) {
  error =
    error ?? showPricesInverted
      ? 'Price must be lower than ' + derivedAuctionInfo?.clearingPrice?.invert().toSignificant(5)
      : 'Price must be higher than ' + derivedAuctionInfo?.clearingPrice?.toSignificant(5)
}
```

or use your logic in both places. Mainttaining two logics is always more work

Reply:
I have seen how it works and I can use it.